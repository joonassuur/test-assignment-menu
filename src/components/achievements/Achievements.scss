.achievements-container {
  display: flex;
  border-top: 1px solid #DCDCE0;
  border-bottom: 1px solid #DCDCE0;
  flex-direction: column;
  align-items: center;
  align-items: flex-start;
  .achievements-sub-container-top {
    display: flex;
  }
  .achievements-info {
    text-align: left;
  }
  .achievements-progress {
    display: flex;
  }

}
