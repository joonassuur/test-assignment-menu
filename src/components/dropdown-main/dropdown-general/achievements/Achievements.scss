.achievements-container {
  display: flex;
  border-top: 1px solid #dcdce0;
  border-bottom: 1px solid #dcdce0;
  flex-direction: column;
  align-items: flex-start;
  .dot {
    margin: 0 5px; 
  }
  .achievements-sub-container-top {
    display: flex;
    align-items: center;
    .settings-icon {
      margin-right: 5px;
    }
  }
  .achievements-info {
    text-align: left;
  }
  .achievements-progress {
    display: flex;
    margin-top: 5px;
    width: 100%;
  }
}
