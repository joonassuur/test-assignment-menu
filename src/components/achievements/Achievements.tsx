import React from "react";
import { useDispatch } from "react-redux";
import { setActiveDD } from "../../redux/Index";
import "./Achievements.scss";

function Achievements() {
  const dispatch = useDispatch();

  const toggleAchievements = () => {
    dispatch(setActiveDD("achievements"));
  };

  return (
    <div className="achievements-container item" onClick={toggleAchievements}>
      <div className="achievements-sub-container-top">
        <div className="settings-icon">
          <span className="material-icons">school</span>
        </div>
        <div className="achievements-info">
          <div className="title">Achievements</div>
          <div className="subtitle">3/11 done * 27%</div>
        </div>
      </div>
      <div className="achievements-progress">
        <progress
          className="progress small-bottom-margin"
          value="27"
          max="100"
        ></progress>
      </div>
    </div>
  );
}

export default Achievements;
