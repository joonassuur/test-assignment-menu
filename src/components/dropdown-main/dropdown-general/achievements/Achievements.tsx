import React from "react";
import { useDispatch } from "react-redux";
import { setActiveDD } from "../../../../redux/Index";
import "./Achievements.scss";

function Achievements() {
  const dispatch = useDispatch();

  const toggleAchievements = () => {
    dispatch(setActiveDD("achievements"));
  };

  return (
    <div className="achievements-container item" onClick={toggleAchievements}>
      <div className="achievements-sub-container-top">
        <div className="settings-icon">
          <span className="material-icons-outlined">school</span>
        </div>
        <div className="achievements-info">
          <div className="title">Achievements</div>
          <div className="subtitle">3/11 done <span className="dot">Â·</span>  27%</div>
        </div>
      </div>
      <div className="achievements-progress">
        <progress
          className="progress"
          value="27"
          max="100"
        ></progress>
      </div>
    </div>
  );
}

export default Achievements;
