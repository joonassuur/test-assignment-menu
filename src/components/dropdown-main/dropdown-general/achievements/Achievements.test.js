import React from "react";
import TestRenderer from "react-test-renderer";

import Achievements from "./Achievements";

const mockDispatch = jest.fn();

jest.mock("react-redux", () => ({
  useSelector: jest.fn((fn) => fn()),
  useDispatch: () => mockDispatch,
}));

jest.mock("../../../../redux/Index");

describe("Achievements component", () => {
  test("Matches the snapshot", () => {
    const achievements = TestRenderer.create(<Achievements />);
    expect(achievements.toJSON()).toMatchSnapshot();
  });
});
