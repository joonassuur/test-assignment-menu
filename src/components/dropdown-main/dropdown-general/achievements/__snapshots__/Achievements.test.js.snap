// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Achievements component Matches the snapshot 1`] = `
<div
  className="achievements-container item"
  onClick={[Function]}
>
  <div
    className="achievements-sub-container-top"
  >
    <div
      className="settings-icon"
    >
      <span
        className="material-icons-outlined"
      >
        school
      </span>
    </div>
    <div
      className="achievements-info"
    >
      <div
        className="title"
      >
        Achievements
      </div>
      <div
        className="subtitle"
      >
        3/11 done 
        <span
          className="dot"
        >
          Â·
        </span>
          27%
      </div>
    </div>
  </div>
  <div
    className="achievements-progress"
  >
    <progress
      className="progress"
      max="100"
      value="27"
    />
  </div>
</div>
`;
